src/core/Akka.Remote/Endpoint.cs

```diff
@@ -528,7 +528,7 @@ protected override SupervisorStrategy SupervisorStrategy()
                 _log.Warning("Association with remote system {0} has failed; address is now gated for {1} ms. Reason is: [{2}]", _remoteAddress, _settings.RetryGateClosedFor.TotalMilliseconds, ex);
                 UidConfirmed = false; // Need confirmation of UID again
 
-                if ((_resendBuffer.Nacked.Any() || _resendBuffer.NonAcked.Any()) && _bailoutAt == null)
+                if ((_resendBuffer.Nacked.Any() || _resendBuffer.NonAcked.Any()) && _bailoutAt == Deadline.Never)
```

Comment:
Check for `_bailloutAt == Deadline.Never` instead of `null`; always reset to `Deadline.Never` instead of `null`.